### PR TITLE
Simple layers control

### PR DESCRIPTION
**Pull request by MariDani**
Creation of a new component Simple Layers Control

- updated style
- custom positioning added to props
- new thumbnails

image

### CODE REVIEW

**Comment by vlach1989** [comment]
The thubnail images are quite big. Try to compress them using https://tinypng.com/ or something similar

File: src/controls/SimpleLayersControl/index.js [hunk 1]
```diff
@@ -100,48 +94,62 @@ class SimpleLayersControl extends React.PureComponent {
...
+		const style = {
 			width: `${width}rem`,
 			height: `${height}rem`,
 			margin: `${margin}rem`,
 		};
-
-		if (layer.thumbnail) {
-			// TODO check type of thumbnail
-			// style.backgroundImage = `url(${require('./img/' + layer.thumbnail + '.png')})`;
+		const previewParam = {};
```
Comment (vlach1989) [inline src/controls/SimpleLayersControl/index.js:120]: IMO let would be better if you manipulate with the variable

Comment (vlach1989) [inline src/controls/SimpleLayersControl/index.js:120]: IMO let would be better if you manipulate with the variable

**Review by vlach1989 — CHANGES_REQUESTED** [review]

**Review by vlach1989 — DISMISSED** [review]

File: src/controls/SimpleLayersControl/index.js [hunk 2]
```diff
@@ -100,48 +94,62 @@ class SimpleLayersControl extends React.PureComponent {
...
-			// TODO check type of thumbnail
-			// style.backgroundImage = `url(${require('./img/' + layer.thumbnail + '.png')})`;
+		let previewParam = {};
+		if (layerTemplate?.data?.thumbnail) {
+			previewParam['src'] =
+				layerTemplate.data.thumbnail in images
+					? images[layerTemplate.data.thumbnail]
+					: images.noPreview;
+			previewParam['alt'] = layerTemplate.data.thumbnail;
 		}
```
Comment (vdubr) [inline src/controls/SimpleLayersControl/index.js:127]: There is no else for case if layerTemplate has not thumbnail. 

Thre should bo some placeholder?

**Comment by vdubr** [comment]
One last question. Images of thumbnail are loaded dynamically or they are stored in base64 in build.js file?

File: src/controls/SimpleLayersControl/index.js (hunk 2 above)
Comment (vlach1989) [reply src/controls/SimpleLayersControl/index.js:130]: https://github.com/gisat-panther/ptr-maps/pull/84/commits/3ec8eb7775634c6957d146e0f68dde1366fef4b5

File: src/controls/SimpleLayersControl/index.js [hunk 3]
```diff
@@ -100,48 +94,62 @@ class SimpleLayersControl extends React.PureComponent {
...
 			margin: `${margin}rem`,
 		};
-
-		if (layer.thumbnail) {
-			// TODO check type of thumbnail
-			// style.backgroundImage = `url(${require('./img/' + layer.thumbnail + '.png')})`;
+		let previewParam = {};
+		if (layerTemplate?.data?.thumbnail) {
+			previewParam['src'] =
+				layerTemplate.data.thumbnail in images
```
[the remark this answers is not in the document]
Comment (vlach1989) [reply src/controls/SimpleLayersControl/index.js]: https://github.com/gisat-panther/ptr-maps/pull/84/commits/3ec8eb7775634c6957d146e0f68dde1366fef4b5

**Comment by vlach1989** [comment]
> One last question. Images of thumbnail are loaded dynamically or they are stored in base64 in build.js file?

Images are stored in code and are linked by their name with layerTemplate.data.thumbnail prop

**Review by vlach1989 — APPROVED** [review]

**Pull request merged**